packages/protocol/contracts/L1/libs/LibProving.sol

```diff
@@ -377,6 +377,24 @@ library LibProving {
                         false
                     );
                 } else {
+                    // Notify the verifier that a proof it verified has been
```

Comment:
I don't understand all the different phases yet of this design, but shouldn't this be done in the `evidence.tier == tran.tier` case? Because if just another proving system has a different result, how can the verifier make a self assessment using that information?

Reply:
The idea is if a proof A on tier 3 is contested by another proof B on the same tier, and a proof C on tier 5 has proven that B is correct and A is not, then we notify the verifier for tier 3,...

Reply:
The example I gave above may not be clear. Let me try this:

1. Alice submits a block header X, along with a tier 2 proof.
2. Bob challenges Alice's claim by presenting a different header Y. No proof is required for Y, and even if provided, the protocol ignores it. This means the challenger can be any user, not necessarily a prover.
3. Charley validates that Y is the correct header hash using a tier 3 proof. Only at this point do we suspect that the tier 2 proof might be flawed. Alice's tier 2 proof was initially verified but is now overridden by Charley's tier 3 proof. The tier 2 verifier is notified and may opt for self-disabling.

Reply:
I see, so there is some subjectiveness that comes with this approach, compared to the two conflicting proofs for the same proof type which always means it's buggy.

Reply:
Yes